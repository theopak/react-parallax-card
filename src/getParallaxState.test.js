import {
  default as makeGetParallaxState,
  getParallaxState,
} from './getParallaxState'

describe('getParallaxState', () => {
  const prevState = {
    height: 90,
    width: 91,
    left: 93,
    top: 92,
  }
  const props = {
    tiltMidpoint: 0.52,
    tiltX: 0.1,
    tiltY: 0.07,
  }

  const mockBody = {
    scrollLeft: 11,
    scrollTop: 12,
  }
  const mockEvent = {
    touches: [{ pageX: 21, pageY: 22 }],
  }

  test('default export is a factory', () => {
    expect(typeof makeGetParallaxState).toEqual('function')
  })

  describe('functionality', () => {
    const testCases = [
      [
        { scrollLeft: 0, scrollTop: 0 },
        { pageX: 21, pageY: 22 },
        { width: 100, height: 100, left: 0, top: 0 },
        { tiltMidpoint: 0.5, tiltX: 0.1, tiltY: 0.1 },
      ],
      [
        { scrollLeft: 0, scrollTop: 0 },
        { touches: [{ pageX: 21, pageY: 22 }] },
        { width: 100, height: 100, left: 0, top: 0 },
        { tiltMidpoint: 0.5, tiltX: 0.1, tiltY: 0.1 },
      ],
      [
        { scrollLeft: 0, scrollTop: 0 },
        { pageX: 8, pageY: 9, touches: [{ pageX: 1, pageY: 2 }] },
        { width: 100, height: 100, left: 0, top: 0 },
        { tiltMidpoint: 0.5, tiltX: 0.1, tiltY: 0.1 },
      ],
      [
        { scrollLeft: 10, scrollTop: 20 },
        { pageX: 21, pageY: 22 },
        { width: 100, height: 100, left: 0, top: 0 },
        { tiltMidpoint: 0.5, tiltX: 0.1, tiltY: 0.1 },
      ],
      [
        { scrollLeft: 10, scrollTop: 20 },
        { pageX: 30, pageY: 40 },
        { width: 250, height: 250, left: 18, top: 19 },
        { tiltMidpoint: 0.45, tiltX: 0.101, tiltY: 0.202 },
      ],
    ]

    test.each(testCases)(
      'getParallaxState(%p, %p, %p, %p)',
      (body, event, prevState, props) => {
        const result = getParallaxState(body, event, prevState, props)
        expect(result).toMatchSnapshot()
      }
    )
  })
})
