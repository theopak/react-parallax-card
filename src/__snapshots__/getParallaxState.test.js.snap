// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getParallaxState functionality getParallaxState({"scrollLeft": 0, "scrollTop": 0}, {"pageX": 8, "pageY": 9, "touches": [Array]}, {"height": 100, "left": 0, "top": 0, "width": 100}, {"tiltMidpoint": 0.5, "tiltX": 0.1, "tiltY": 0.1}) 1`] = `
Object {
  "angle": -225.59065721464668,
  "offsetX": 0.49,
  "offsetY": 0.48,
  "rotateX": -15.513599999999999,
  "rotateY": 15.836800000000004,
  "scale": 1.07,
  "shineMidpointAlpha": 0.008,
}
`;

exports[`getParallaxState functionality getParallaxState({"scrollLeft": 0, "scrollTop": 0}, {"pageX": 21, "pageY": 22}, {"height": 100, "left": 0, "top": 0, "width": 100}, {"tiltMidpoint": 0.5, "tiltX": 0.1, "tiltY": 0.1}) 1`] = `
Object {
  "angle": -226.00508600525419,
  "offsetX": 0.29000000000000004,
  "offsetY": 0.28,
  "rotateX": -9.0496,
  "rotateY": 9.372800000000002,
  "scale": 1.07,
  "shineMidpointAlpha": 0.08800000000000001,
}
`;

exports[`getParallaxState functionality getParallaxState({"scrollLeft": 0, "scrollTop": 0}, {"touches": [Array]}, {"height": 100, "left": 0, "top": 0, "width": 100}, {"tiltMidpoint": 0.5, "tiltX": 0.1, "tiltY": 0.1}) 1`] = `
Object {
  "angle": -226.00508600525419,
  "offsetX": 0.29000000000000004,
  "offsetY": 0.28,
  "rotateX": -9.0496,
  "rotateY": 9.372800000000002,
  "scale": 1.07,
  "shineMidpointAlpha": 0.08800000000000001,
}
`;

exports[`getParallaxState functionality getParallaxState({"scrollLeft": 10, "scrollTop": 20}, {"pageX": 21, "pageY": 22}, {"height": 100, "left": 0, "top": 0, "width": 100}, {"tiltMidpoint": 0.5, "tiltX": 0.1, "tiltY": 0.1}) 1`] = `
Object {
  "angle": -219.0938588862295,
  "offsetX": 0.39,
  "offsetY": 0.48,
  "rotateX": -15.513599999999999,
  "rotateY": 12.604800000000001,
  "scale": 1.07,
  "shineMidpointAlpha": 0.008,
}
`;

exports[`getParallaxState functionality getParallaxState({"scrollLeft": 10, "scrollTop": 20}, {"pageX": 30, "pageY": 40}, {"height": 250, "left": 18, "top": 19, "width": 250}, {"tiltMidpoint": 0.45, "tiltX": 0.101, "tiltY": 0.202}) 1`] = `
Object {
  "angle": -224.7680345487018,
  "offsetX": 0.442,
  "offsetY": 0.446,
  "rotateX": -16.08837888,
  "rotateY": 31.91716352,
  "scale": 1.07,
  "shineMidpointAlpha": 0.0016,
}
`;
